src/main/java/seedu/bankwithus/user/Account.java

```diff
@@ -75,4 +82,16 @@ public String getName() {
     public SaveGoal getSaveGoal() {
         return this.saveGoal;
     }
+
+    //@@author Sherlock-YH
+    public boolean isMoreThanTwoDecimal(float num) {
```

Comment:
you can add this isMoreThanTwodecimal() method to AccountList class as well so that u can directly do:
```
if (isMoreThanTwoDecimal(balance)){
                throw new UserInputAmountNotValid();
}
```
            
 in that way you will not need to access the .getMainAccount method and can resolve the issue mentioned below of an index out of bounds exceptions.